packages/notification/core/notification.ts

```diff
@@ -0,0 +1,46 @@
+import {
+  Renderable,
+  Notification,
+  NotificationOptions,
+  ValueOrFunction,
+} from '../types'
+import { genId } from './utils'
+import { dispatch, ActionType } from './store'
+
+type Message = ValueOrFunction<Renderable, Notification>
+
+type NotificationHandler = (
+  message: Message,
+  options?: NotificationOptions
+) => string
+
+const createNotification = (message: Message, options): Notification => ({
+  createdAt: Date.now(),
+  visible: true,
+  message,
+  pauseDuration: 0,
+  ...options,
+  id: options?.id || `notification-${genId()}`,
+})
+
+const createHandler = (): NotificationHandler => (message, options) => {
+  const notification = createNotification(message, options)
+  dispatch({ type: ActionType.UPSERT_NOTIFICATION, notification })
+  return notification.id
+}
+
+const notification = (message: Message, options?: NotificationOptions) => {
+  createHandler()(message, options)
+}
```

Comment:
This setup allows us to create methods on `notification` to do things like remove and dismiss notifications.

Below you can see how this is being used. 

```diff
notification((n) => {
  return (
    <Notification
      description="lAliquet in lacus, scelerisque tristique purus. Lob ortis felis orci lacus egestas et turpis."
      cta={{
        title: 'Standalone link',
        url: '/'
      }}
+      onDismiss={() => notification.remove(n.id)}
    />
  )
})
```

In the future when we offer the different notification types. I can envision an api for generating the type of notification via these methods.

```js
// Generate a notification as a toast
notfication.toast((n) => {})

// Generate a notification as a non-modal dialog
notfication.dialog((n) => {})
```

Reply:
⛏️ nit: Ah it looks like in `react-hot-toast` `createHandler` accepts a type: https://github.com/timolins/react-hot-toast/blob/main/src/core/toast.ts#L35. Are we thinking we would bring back that capability to enable different types of notifications at this level? Otherwise if we're only using `createHandler` once like this I agree with Ashlee it might be clearer if `createHandler` was inlined into the definition of `notification`

Reply:
@BRKalow yes, I simplified for now, but I did imagine once we understand a bit more how we'd introduce the different notification needs, that we'd have notification types "toast" and "dialog" which would then be treated accordingly when rendering.